cola/widgets/dag.py

```diff
@@ -585,6 +594,14 @@ def histories_selected(self, histories):
         self.revtext.setText(text)
         self.display()
 
+    def difftool_selected(self, files):
+        bottom, top = self.treewidget.selected_commit_range()
+        if not top:
+            return
+        if not bottom:
+            bottom = '%s^' % top
+        difftool.launch([bottom, top, '--'] + list(files))
```

Comment:
Thanks for looking into this feature. I think we can simplify this part right here a bit.. if the intention is that we can choose any commit and run difftool on it, then we probably shouldn't be asking the widget for the "below" item.

I think it would be simpler to just get the selected item's sha1.  difftool can then be launched using just that sha1 and git's magical "^!" operator, which means, "diff against the commit's parent".

```python

    item = self.selected_item()
    if not item:
        return
    diff_expr = item.commit.sha1 + '^!'
    difftool.launch([diff_expr, '--'] + list(files))
```

Would you mind giving that a try?  I think it should allow us to do this without needing to implement a separate `selected_commit_range()`.